lib/Doctrine/DBAL/Migrations/Provider/LazySchemaDiffProvider.php

```diff
@@ -32,8 +33,23 @@ class LazySchemaDiffProvider implements SchemaDiffProviderInterface
     /** @var SchemaDiffProviderInterface */
     private $originalSchemaManipulator;
 
-    public function __construct(LazyLoadingValueHolderFactory $proxyFactory, SchemaDiffProviderInterface $originalSchemaManipulator)
+    /** @var  Configuration */
+    private $proxyConfig;
+
+    public function __construct(LazyLoadingValueHolderFactory $proxyFactory=null, SchemaDiffProviderInterface $originalSchemaManipulator)
     {
+        if ($proxyFactory === null ) {
+            $this->proxyConfig = new Configuration();
+            $proxyFolder = __DIR__ . DIRECTORY_SEPARATOR . 'SchemaProxy';
+            if (!is_dir($proxyFolder)) {
```

Comment:
Instead of relying on the filesystem, given that this is a rarely-used feature, I suggest using the [`EvaluatingGeneratorStrategy`](https://github.com/Ocramius/ProxyManager/blob/57e9272ec0e8deccf09421596e0e2252df440e11/src/ProxyManager/GeneratorStrategy/EvaluatingGeneratorStrategy.php#L29)  instead, which will not write file to disk (and save you a huge amount of headaches). You will also not need the custom autoloader in that case